temba/channels/types/twitter/tasks.py

```
@@ -0,0 +1,77 @@
+from __future__ import print_function, unicode_literals
+
+from celery.task import task
+from django_redis import get_redis_connection
+from twython import Twython
+
+from temba.contacts.models import ContactURN, URN, TWITTER_SCHEME
+from temba.utils import chunk_list
+from django.conf import settings
+
+
+@task(track_started=True, name='resolve_twitter_ids_task')
+def resolve_twitter_ids():
+    r = get_redis_connection()
+
+    # TODO: we can't use our non-overlapping task decorator as it creates a loop in the celery resolver when registering
+    if r.get('resolve_twitter_ids_task'):  # pragma: no cover
```

I fought that problem separately.. the behavior with this one is actually a total hang of Celery, I pdb'ed it a bit and strangely it does it when trying to check type of the task with an instanceof. Don't know if there is some kind of circular aspect to it perhaps that then isn't caught? Celery likes failing silently.